src/eth/executor.rs

```diff
@@ -223,6 +265,26 @@ impl EthExecutor {
     }
 }
 
+fn cmp_execution_with_receipt(execution: &Execution, receipt: &ExternalReceipt) {
```

Comment:
The purpose of the executor is just to orchestrate different components (EVM, Miner, Storage, etc).

I think it should be easier to maintain and test if we isolate the checks in some struct.

Example:
ExecutionChanges receives ExternalReceipt and compares current state with expected receipt state and logs.
ExecutionChanges receives ExternalBlock and compare other necessary stuff.

Reply:
Extracted the comparison functions to Block and Execution. As a side-effect I had to add a clone [here](https://github.com/cloudwalk/stratus/blob/89c7fc0ad9b938a08d71b43a0ac657cab4bc9c9b/src/eth/executor.rs#L103)